### ISSUE TITLE
custom onCreate.fields throws exception when trying to read

### ISSUE THREAD

**Issue by red-0ne** [orientechnologies/orientdb#4886]
When executing the following sequence:

``` sql

orientdb> CREATE DATABASE remote:localhost/test root root memory

Creating database [remote:localhost/test] using the storage type [memory]...
Connecting to database [remote:localhost/test] with user 'admin'...OK
Database created successfully.

Current database is: remote:localhost/test
orientdb {db=test}> CREATE CLASS test EXTENDS ORestricted

Class created successfully. Total classes in database now: 11

orientdb {db=test}> ALTER CLASS test CUSTOM onCreate.fields=_allowRead

Class updated successfully

orientdb {db=test}> DISCONNECT

Disconnecting from the database [test]...OK
orientdb> CONNECT remote:localhost/test writer writer

Connecting to database [remote:localhost/test] with user 'writer'...OK
orientdb {db=test}> INSERT INTO test SET name = 'a'

Inserted record 'test#11:0{name:a,_allowRead:[1]} v1' in 0.005000 sec(s).

orientdb {db=test}> SELECT FROM test


0 item(s) found. Query executed in 0.007 sec(s).
```

Owner cannot read the record, and we get this on the server log:

```
2015-08-30 08:49:52:536 INFO  Dropped database 'memory:test' [ONetworkProtocolBinary]
2015-08-30 08:49:53:423 INFO  {db=test} Created database 'test' of type 'memory' [ONetworkProtocolBinary]{db=test} Error on fetching record during browsing. The record has been skipped
com.orientechnologies.orient.core.exception.ODatabaseException: Error on retrieving record #11:0 (cluster: test)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1765)
        at com.orientechnologies.orient.core.tx.OTransactionNoTx.loadRecord(OTransactionNoTx.java:99)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:1562)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.load(ODatabaseDocumentTx.java:119)
        at com.orientechnologies.orient.core.iterator.OIdentifiableIterator.readCurrentRecord(OIdentifiableIterator.java:292)
        at com.orientechnologies.orient.core.iterator.ORecordIteratorClusters.begin(ORecordIteratorClusters.java:301)
        at com.orientechnologies.orient.core.iterator.ORecordIteratorClusters.setRange(ORecordIteratorClusters.java:81)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbstract.searchInClasses(OCommandExecutorSQLResultsetAbstract.java:479)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbstract.searchInClasses(OCommandExecutorSQLResultsetAbstract.java:468)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.searchInClasses(OCommandExecutorSQLSelect.java:885)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLResultsetAbstract.assignTarget(OCommandExecutorSQLResultsetAbstract.java:209)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.assignTarget(OCommandExecutorSQLSelect.java:491)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.executeSearch(OCommandExecutorSQLSelect.java:473)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLSelect.execute(OCommandExecutorSQLSelect.java:449)
        at com.orientechnologies.orient.core.sql.OCommandExecutorSQLDelegate.execute(OCommandExecutorSQLDelegate.java:67)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.executeCommand(OAbstractPaginatedStorage.java:1610)
        at com.orientechnologies.orient.core.storage.impl.local.OAbstractPaginatedStorage.command(OAbstractPaginatedStorage.java:1566)
        at com.orientechnologies.orient.core.sql.query.OSQLQuery.run(OSQLQuery.java:78)
        at com.orientechnologies.orient.core.sql.query.OSQLSynchQuery.run(OSQLSynchQuery.java:85)
        at com.orientechnologies.orient.core.query.OQueryAbstract.execute(OQueryAbstract.java:33)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.command(ONetworkProtocolBinary.java:1354)
        at com.orientechnologies.orient.server.network.protocol.binary.ONetworkProtocolBinary.executeRequest(ONetworkProtocolBinary.java:402)
        at com.orientechnologies.orient.server.network.protocol.binary.OBinaryNetworkProtocolAbstract.execute(OBinaryNetworkProtocolAbstract.java:223)
        at com.orientechnologies.common.thread.OSoftThread.run(OSoftThread.java:77)
Caused by: java.lang.NullPointerException
        at com.orientechnologies.orient.core.metadata.security.OSecurityShared.isAllowed(OSecurityShared.java:201)
        at com.orientechnologies.orient.core.metadata.security.OSecurityProxy.isAllowed(OSecurityProxy.java:44)
        at com.orientechnologies.orient.core.metadata.security.ORestrictedAccessHook.isAllowed(ORestrictedAccessHook.java:129)
        at com.orientechnologies.orient.core.metadata.security.ORestrictedAccessHook.onRecordBeforeRead(ORestrictedAccessHook.java:85)
        at com.orientechnologies.orient.core.hook.ODocumentHookAbstract.onTrigger(ODocumentHookAbstract.java:242)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.callbackHooks(ODatabaseDocumentTx.java:1021)
        at com.orientechnologies.orient.core.db.document.ODatabaseDocumentTx.executeReadRecord(ODatabaseDocumentTx.java:1747)
        ... 23 more
```

OrientDB 2.2.0-SNAPSHOT (ae2bbcb)

**Comment by red-0ne** [comment]
Any chance to get this regression handled before 2.2 is out?

**Comment by luigidellaquila** [comment]
Hi @red-0ne 

thanks for reporting. I tagged it to be addressed in 2.2

**Comment by red-0ne** [comment]
Could you please take a look at this regression? It is completely preventing us from using record level security while developing against 2.2 branch.

Thank you.
